src/pages/Home/Home.page.jsx

```diff
@@ -1,23 +1,58 @@
-import React, { useRef } from 'react';
+import React, { useRef, useState, useEffect } from 'react';
 import { Link, useHistory } from 'react-router-dom';
 
 import { useAuth } from '../../providers/Auth';
 import './Home.styles.css';
+import VideoList from '../../components/VideoList';
+import mockVideos from '../../youtube-videos-mock.json';
+import { useAppContext } from '../../state/AppProvider';
 
 function HomePage() {
   const history = useHistory();
   const sectionRef = useRef(null);
   const { authenticated, logout } = useAuth();
+  const [isLoading, setIsLoading] = useState(true);
+  const [videos, setVideos] = useState(true);
+  const { state } = useAppContext();
 
   function deAuthenticate(event) {
     event.preventDefault();
     logout();
     history.push('/');
   }
 
+  // const YOUTUBE_SEARCH_ENDPOINT =
+  //   'https://youtube.googleapis.com/youtube/v3/search?part=snippet&maxResults=2&q=';
+
+  const fetchVideos = async () => {
+    try {
+      console.log('Querying ', state.searchValue, '...');
+      throw Error('Not using Youtube API for search');
+      // setIsLoading(true);
+      // const res = await fetch(
+      //   `${YOUTUBE_SEARCH_ENDPOINT}${state.searchValue}&key=${process.env.REACT_APP_YOUTUBE_API_KEY}`
+      // );
+      // if (!res.ok) throw Error('Youtube API response ', res.status);
+      // const data = await res.json();
+      // setVideos(data);
+      // setIsLoading(false);
```

Comment:
Here is one place where you should use the youtube API, the quota has an allocation of 10,000 units per day, here is a list of some quota costs:

- A read operation that retrieves a list of resources -- channels, videos, playlists -- usually costs 1 unit.
- A write operation that creates, updates, or deletes a resource usually has costs 50 units.
- A search request costs 100 units.
- A video upload costs 1600 units.

You can read more about it [here](https://developers.google.com/youtube/v3/getting-started)

When you exceed the limit it will be only for that day, the next day you will have another 10000 units available to use, to prevent that you can save the response and use the mocked data while developing

The acceptance criteria for previous challenge is:

## Acceptance Criteria
- [ ] Videos in the Home View are fetched from the YouTube API and displayed correctly.
- [ ] Users can search for YouTube videos using the search field on the header.
- [x] A video can be played from the **Video Details View** after clicking on it.
- [ ] The video information and related videos list are displayed correctly on the **Video Details View**.
- [x] When a user clicks on a related video the video player, information and related videos list are updated.

### Bonus Points
- [ ] Custom Hooks for API calls are implemented and tested correctly.
- [ ] Testing coverage is above 60%. (Please include a screenshot of the code coverage report in your PR description).

For the next challenge use the youtube API, so I can mark your second deliverable as completed, you have the skeleton so plugging in the API shouldn't be hard

Reply:
Sounds good, thanks, I still can't find why my quota was exceeded so quick but I will check